src/Microsoft.SqlTools.ManagedBatchParser/BatchParser/Parser.cs

```diff
@@ -614,10 +614,7 @@ internal void RaiseError(ErrorCode errorCode, string message = null)
 
         internal static void RaiseError(ErrorCode errorCode, Token token, string message = null)
         {
-            if (message == null)
-            {
-                message = string.Format(CultureInfo.CurrentCulture, SR.BatchParser_IncorrectSyntax, token.Text);
-            }
+            message ??= string.Format(CultureInfo.CurrentCulture, SR.BatchParser_IncorrectSyntax, token.Text);
```

Comment:
>??= [](http://example.com/codeflow?start=20&length=3)

Glad to learn about this new operator, but... isn't this code just a 1-liner:

```csharp
throw new BatchParserException(errorCode, token, message ?? string.Format(CultureInfo.CurrentCulture, SR.BatchParser_IncorrectSyntax, token.Text));
```

Reply:
Now it is :)